src/main/java/com/minecolonies/coremod/colony/crafting/CustomRecipe.java

```diff
@@ -0,0 +1,232 @@
+package com.minecolonies.coremod.colony.crafting;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.colony.requestsystem.StandardFactoryController;
+import com.minecolonies.api.crafting.IRecipeStorage;
+import com.minecolonies.api.research.effects.IResearchEffect;
+import com.minecolonies.api.util.constant.TypeConstants;
+import com.minecolonies.coremod.research.UnlockAbilityResearchEffect;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.Blocks;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.registries.ForgeRegistries;
+
+import java.util.ArrayList;
+
+/**
+ * This class represents a recipe loaded from custom data that is available to a crafter
+ * but not to a player
+ */
+public class CustomRecipe
+{
+
+    /**
+     * The property name that indicates type for the recipe
+     */
+    public static final String RECIPE_TYPE = "type";
```

Comment:
maybe call the property ones RECIPE_TYPE_PROP, so the next one doesn't have such a confusing name =D